src/texthooks/fix_unicode_dashes.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+"""
+A fixer script which crawls text files and replaces Unicode dash characters
+with ASCII equivalents.
+
+This fixes copy-paste from some applications which replace hyphens with
+Unicode dash characters like en-dash (–), em-dash (—), minus sign (−),
+and fullwidth hyphen (－).
+
+The following Unicode characters are replaced:
+- U+2013 EN DASH (–) → ASCII hyphen (-)
+- U+2014 EM DASH (—) → ASCII double hyphen (--)
+- U+2212 MINUS SIGN (−) → ASCII hyphen (-)
+- U+FF0D FULLWIDTH HYPHEN-MINUS (－) → ASCII hyphen (-)
+
+In files with the offending characters, they are replaced and the run is
+marked as failed. This makes the script suitable as a pre-commit fixer.
+"""
+import re
+import sys
+
+from ._common import all_filenames, codepoints2chars, parse_cli_args
+from ._recorders import DiffRecorder
+
+
+def codepoints2regex(codepoints):
+    return re.compile("(" + "|".join(codepoints2chars(codepoints)) + ")")
+
+
+# Unicode codepoints for dash characters that should be replaced with ASCII hyphens
+DEFAULT_HYPHEN_CODEPOINTS = (
+    "2013",  # EN DASH (–)
+    "2212",  # MINUS SIGN (−)
+    "FF0D",  # FULLWIDTH HYPHEN-MINUS (－)
```

Comment:
I think this character set could also include
```
2012  # Figure Dash
02D7  # Modifier Letter Minus Sign
2796  # Heavy Minus Sign
2010  # Hyphen
2011  # Non-Breaking Hyphen
FE63  # Small Hyphen-Minus
```

Do any (all) of these appeal?
(I think the Mongolian Todo Soft Hyphen is a bit too obscure to include out of the box.)

I also would consider whether or not `FF0D` should be treated as an em-dash and replaced with `--` when ASCII-izing text.